packages/labs/nextjs/src/index.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {NextConfig} from 'next';
+
+// TODO(augustjk) Consider adding option for adding template shadowroot polyfill
+// automatically
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface LitSsrPluginOptions {}
+
+export = (_pluginOptions: LitSsrPluginOptions = {}) =>
+  (nextConfig: NextConfig = {}) => {
+    return Object.assign({}, nextConfig, {
+      webpack: (config, options) => {
+        const {isServer, nextRuntime, webpack} = options;
+        config.module.rules.unshift({
+          // Grab entry points for all pages
+          // TODO(augustjk) This may not work for the new "app" directory
+          // https://github.com/lit/lit/issues/3657
+          test: /\/pages\//,
```

Comment:
Would it make sense to test for `ts` and `tsx` files only here. We are using SCSS modules from nextjs which leads to scss files under our pages directory (https://nextjs.org/docs/basic-features/built-in-css-support#sass-support) and the loader seems to break these imports. Restricting the test to ts and tsx files only here fixes the issue for us.

Reply:
Thank you for catching and reporting that! Added extensions js,jsx,ts,tsx to the test.